### PR TITLE
[4.1] Refactor vue browser items

### PR DESCRIPTION
**Pull request by laoneo**
Followup pr of #35332.

### Summary of Changes
As #35332 gets abandoned, this one here is a conflict fix. All the old commits should be preserved.

More information can be found in the pr #35332 by @JenSeReal.

### Testing Instructions
Open media manager and execute the actions and browse around.

### Actual result BEFORE applying this Pull Request
All should work.

### Expected result AFTER applying this Pull Request
All should work.

### CODE REVIEW

File: administrator/components/com_media/resources/scripts/components/browser/actionItems/actionItemsContainer.vue [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    downloadable: { type: Boolean, default: false },
+    shareable: { type: Boolean, default: false },
+  },
+  data() {
+    return {
+      showActions: false,
+    };
+  },
+  computed: {
+    /* Check if the item is an document to edit */
```
Comment (brianteeman) [inline administrator/components/com_media/resources/scripts/components/browser/actionItems/actionItemsContainer.vue:134]:
```suggestion
    /* Check if the item is a document to edit */
```

File: administrator/components/com_media/resources/scripts/components/browser/actionItems/actionItemsContainer.vue [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    canEdit() {
+      return this.editable();
+    },
+  },
+  methods: {
+    /* Hide actions dropdown */
+    /* Hide actions dropdown */
```
Comment (brianteeman) [inline administrator/components/com_media/resources/scripts/components/browser/actionItems/actionItemsContainer.vue]:
```suggestion
```

File: administrator/components/com_media/resources/scripts/components/browser/actionItems/actionItemsContainer.vue [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    hideActions() {
+      this.$refs.container.hideActions();
+    },
+    /* Preview an item */
+    openPreview() {
+      this.$refs.container.openPreview();
+    },
+    /* Preview an item */
```
Comment (brianteeman) [inline administrator/components/com_media/resources/scripts/components/browser/actionItems/actionItemsContainer.vue]:
```suggestion
    /* Download an item */
```

File: administrator/components/com_media/resources/scripts/components/browser/items/image.vue [hunk 4]
```diff
@@ -20,254 +17,74 @@
...
   methods: {
-    /* Preview an item */
-    openPreview() {
-      this.$store.commit(types.SHOW_PREVIEW_MODAL);
-      this.$store.dispatch('getFullContents', this.item);
-    },
-    /* Preview an item */
-    download() {
-      this.$store.dispatch('download', this.item);
+    /* Check if the item is an document to edit */
```
Comment (brianteeman) [inline administrator/components/com_media/resources/scripts/components/browser/items/image.vue]:
```suggestion
    /* Check if the item is a document to edit */
```

**Comment by dgrammatiko** [comment]
@laoneo can you please run lint/fix? `npm run lint:js -- --fix` should do the trick

**Comment by richard67** [comment]
@laoneo System test is failing, I've tried to restart several times now but it always fails at the same place, so maybe it's related to this PR: https://ci.joomla.org/joomla/joomla-cms/48082/1/22

It always fails at the test "Test that it is possible to navigate to a subfolder using double click.".

The screenshot from the test doesn't show anything special: https://ci.joomla.org/artifacts/joomla/joomla-cms/4.1-dev/35887/system-tests/48082

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 3379fccdffb033d6794d9e7e355762b5d1a7c2d0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35887">issues.joomla.org/tracker/joomla-cms/35887</a>.</sub>

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 3379fccdffb033d6794d9e7e355762b5d1a7c2d0<br /><br />Tested as a user, without inspecting the code and English language. From a uses point of view the PR works good, no errors or notices found. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35887">issues.joomla.org/tracker/joomla-cms/35887</a>.</sub>

**Comment by laoneo** [comment]
RTC

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35887">issues.joomla.org/tracker/joomla-cms/35887</a>.</sub>

**Comment by richard67** [comment]
Note for maintainers: When merging with a squash commit, don't remove the "co-authored by" tags in the commit description generated by GitHub so that the author of the original PR is not forgotten.

**Comment by wilsonge** [comment]
Can someone do some quick a11y validation on this one. I assume as we still have the unordered list it's ok. But some of the html here was refactored from divs to ul's originally for proper screenreader compat

**Comment by laoneo** [comment]
@brianteeman can you have a look on the concern from @wilsonge?

**Comment by brianteeman** [comment]
Please ask the accessibility team

**Comment by richard67** [comment]
I will ping them on Glip.

**Comment by laoneo** [comment]
@richard67 thanks

**Comment by dgrammatiko** [comment]
> Can someone do some quick a11y validation on this one.

@wilsonge there are no structural changes here. The same HTML markup just moved around. (or am I wrong?)

**Comment by richard67** [comment]
@dgrammatiko The thing is that while this PR was in progress, changes on the code which is moved around by this PR have been made, and so there were conflicts which were resolved. I think George just wants to be sure that the other changes have been applied to this PR when the conflicts were solved.

**Comment by laoneo** [comment]
The conflicts were created by me as by #35451. I'v fixed them already. There were no structural changes involved in the other pr. I had a second look here and indeed, there are no structural changes as far as I can see, it was before also a list.

**Comment by carcam** [comment]
I can confirm there are no html structural changes from an a11y point of view. Whenever there should be a list, there is a list and no divs are involved. Also with this refactoring, checking the code and A11y fixes will be simpler as we will split longer code in vue components reusing the markup.

I do think this is a RTC!!

**Comment by richard67** [comment]
Is is RTC already.

**Comment by laoneo** [comment]
Thanks @carcam !!

**Comment by bembelimen** [comment]
Thx

**Pull request merged**